core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,13 +527,15 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
-    override def hasData = true
-
-    override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+    override def hasData : Boolean = {
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+      }
 
+      recordIter.hasNext
+    }
+
+    override def receive() : BaseConsumerRecord = {
```

Comment:
@xiaotao183 Thanks for looking into it. I agree that it's probably better not to complicate this patch if the refactor is a bit complicated.